trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/memory/ApplicationUsedMemoryDataFormatterTest.java

```diff
@@ -0,0 +1,48 @@
+package io.bitrise.trace.data.management.formatter.memory;
+
+import org.junit.Test;
+
+import io.bitrise.trace.data.collector.memory.ApplicationUsedMemoryDataCollector;
+import io.bitrise.trace.data.dto.Data;
+import io.bitrise.trace.data.dto.FormattedData;
+import io.bitrise.trace.data.management.formatter.BaseDataFormatterTest;
+import io.opencensus.proto.metrics.v1.LabelKey;
+import io.opencensus.proto.metrics.v1.MetricDescriptor;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for {@link ApplicationUsedMemoryDataFormatter}.
+ */
+public class ApplicationUsedMemoryDataFormatterTest extends BaseDataFormatterTest {
+
+    final ApplicationUsedMemoryDataFormatter formatter = new ApplicationUsedMemoryDataFormatter();
+    final Long memoryUsed = 1234L;
+
+    @Test
+    public void formatData_contentNull() {
+        final Data data = new Data(ApplicationUsedMemoryDataCollector.class);
+        data.setContent(null);
+        assertArrayEquals(new FormattedData[]{}, formatter.formatData(data));
+    }
+
+    @Test
+    public void handleMemoryFormatting() {
```

Comment:
for this test ApplicationUsedMemory and SystemMemory - we already have a MemoryFormatterTest that ensures the Metric created is correct - so these just needed to make sure they set the correct metric descriptor. Does that make sense?